tests/end_to_end/test_g2p.py

```diff
@@ -0,0 +1,530 @@
+        self.assertEqual(1, len(response.associations[0].evidence))
+        evidence = response.associations[0].evidence[0]
+        self.assertEqual('decreased_sensitivity', evidence.description)
+
+    def testGenotypePhenotypeSearchEnsureEnvironment(self):
+        request = protocol.SearchGenotypePhenotypeRequest()
+        request.phenotype_association_set_id = \
+            self.getPhenotypeAssociationSetId()
+        datasetName, featureSet = self.getCGDDataSetFeatureSet()
+        featureId = \
+            "http://ohsu.edu/cgd/27d2169c"
+        obfuscated = self.getObfuscatedFeatureCompoundId(
+            datasetName, featureSet.name, featureId)
+        request.feature_ids.extend([obfuscated])
+        eq = protocol.EvidenceQuery()
+        eq.description = "imatinib"
+        request.evidence.extend([eq])
+        response = self.sendSearchRequest(
+            '/genotypephenotypes/search',
+            request,
+            protocol.SearchGenotypePhenotypeResponse)
+        self.assertEqual(
+            1, len(response.associations[0].environmental_contexts))
+        environmentalContext = response.associations[0] \
+                                       .environmental_contexts[0]
+        self.assertEqual('imatinib', environmentalContext.description)
+
+    def testGenotypeSearchFeaturePagingOne(self):
+        """
+        If page size is set to 1 only one association should be returned
+        """
+        request = protocol.SearchFeaturesRequest()
+        request.page_size = 1
+        datasetName, featureSet = self.getCGDDataSetFeatureSet()
+        request.feature_set_id = featureSet.id
+        request.name = \
+            "KIT *wild"
+        postUrl = "features/search"
+        response = self.sendSearchRequest(
+            postUrl,
+            request,
+            protocol.SearchFeaturesResponse)
+        self.assertEqual(1, len(response.features))
+        self.assertIsNotNone(response.next_page_token)
+
+    def testGenotypeSearchFeaturePagingMore(self):
+        """
+        If page size is not set to more than one association should be returned
+        """
+        request = protocol.SearchFeaturesRequest()
+        datasetName, featureSet = self.getCGDDataSetFeatureSet()
+        request.feature_set_id = featureSet.id
+        request.name = \
+            "KIT *wild"
+        postUrl = "features/search"
+        response = self.sendSearchRequest(
+            postUrl,
+            request,
+            protocol.SearchFeaturesResponse)
+        self.assertGreater(len(response.features), 1)
+        self.assertEqual(response.next_page_token, '')
+
+    def testGenotypeSearchFeaturePagingAll(self):
+        """
+        Loop through all pages
+        """
+        request = protocol.SearchFeaturesRequest()
+        datasetName, featureSet = self.getCGDDataSetFeatureSet()
+        request.feature_set_id = featureSet.id
+        request.page_size = 1
+        request.name = \
```

Comment:
seems like a lot of repetition here... this setup could be refactored into a commonly-used method